sdk/openai/Azure.AI.OpenAI.Assistants/tests/AssistantsTests.cs

```diff
@@ -0,0 +1,297 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Azure.Core.TestFramework;
+using NUnit.Framework;
+
+namespace Azure.AI.OpenAI.Assistants.Tests;
+
+public class AssistantsTests : AssistantsTestBase
+{
+    public AssistantsTests(bool isAsync)
+        : base(isAsync) // , RecordedTestMode.Live)
+    {
+    }
+
+    [RecordedTest]
+    [TestCase(OpenAIClientServiceTarget.NonAzure)]
+    [TestCase(OpenAIClientServiceTarget.Azure)]
+    public async Task CanListAssistants(OpenAIClientServiceTarget target)
+    {
+        AssistantsClient client = GetTestClient(target);
+        Response<PageableList<Assistant>> listResponse = await client.GetAssistantsAsync();
```

Comment:
It could be useful to have some mock tests for the generic `PageableList` type (e.g. to confirm that sorting and pagination work correctly).

Reply:
I believe we're fully beholden to whatever the service gives us, unfortunately; the list operations just treat the parameterization as request input data and we deserialize whatever the response has in it. It'd be good for us to have coverage that validates we deserialize known and expected items (e.g. if a wire list response has items A through E in sequential order, that's what the list should have), but that seems to fit into a slightly different test category given we'd need a static test recording vs. a recordable/updateable one against a persistent and ever-changing service entity.

I'd like to explore how we can improve coverage (here and elsewhere) but wouldn't want to block further if it seems non-trivial (which I think it is).